src/status_im2/contexts/chat/lightbox/view.cljs

```diff
@@ -86,160 +34,88 @@
+                                             :screen-width     screen-width
+                                             :window-height    window-height
+                                             :window-width     window-width
+                                             :props            props}
+         :horizontal                        horizontal?
+         :inverted                          inverted?
+         :paging-enabled                    true
+         :get-item-layout                   (fn [_ index] (get-item-layout _ index item-width))
+         :viewability-config                {:view-area-coverage-percent-threshold 50
+                                             :wait-for-interaction                 true}
+         :shows-vertical-scroll-indicator   false
+         :shows-horizontal-scroll-indicator false
+         :on-viewable-items-changed         callback}]]]
+     (when (and (not @transparent?) (not landscape?))
+       [:f> bottom-view/bottom-view messages index scroll-index insets animations derived
+        item-width props])]))
 
-(defn drag-gesture
-  [{:keys [pan-x pan-y background-color opacity layout]} x? set-full-height?]
-  (->
-    (gesture/gesture-pan)
-    (gesture/enabled true)
-    (gesture/max-pointers 1)
-    (gesture/on-start #(reset! set-full-height? false))
-    (gesture/on-update (fn [e]
-                         (let [translation (if x? (oget e "translationX") (oget e "translationY"))
-                               progress    (Math/abs (/ translation drag-threshold))]
-                           (anim/set-val (if x? pan-x pan-y) translation)
-                           (anim/set-val opacity (- 1 progress))
-                           (anim/set-val layout (* progress -20)))))
-    (gesture/on-end (fn [e]
-                      (if (> (Math/abs (if x? (oget e "translationX") (oget e "translationY")))
-                             drag-threshold)
-                        (do
-                          (anim/animate background-color "rgba(0,0,0,0)")
-                          (anim/animate opacity 0)
-                          (rf/dispatch [:navigate-back]))
-                        (do
-                          #(reset! set-full-height? true)
-                          (anim/animate (if x? pan-x pan-y) 0)
-                          (anim/animate opacity 1)
-                          (anim/animate layout 0)))))))
+(defn- f-lightbox
+  []
+  (let [{:keys [messages index]} (rf/sub [:get-screen-params])
+        props                    (utils/init-props)
+        state                    (utils/init-state messages index)
+        animations               (utils/init-animations)
+        derived                  (utils/init-derived-animations animations)
+        callback                 (fn [e]
+                                   (on-viewable-items-changed e props state))]
+    (anim/animate (:background-color animations) "rgba(0,0,0,1)")
+    (reset! (:data state) messages)
```

Comment:
Totally agree, very acurate comments, since `2` I think the component is not working properly :thinking:

Reply:
Thanks for your comments @ilmotta @ulisesmac 

1. Moving `(reset! (:data state) messages)` inside a `use-effect` will cause it not to work properly. That way it works properly. Idk if there is a better way to do it. If you have suggestions let me know.
2. I believe atoms are initialized outside the rendered. Here we have `lightbox-content` inside `f-lightbox` inside `lightbox`. Take a look at the [current implementation](https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/chat/lightbox/view.cljs). There are 2 functional components (not one) inside the `defn` function.
3. I added back the comment explaining why:
> The initial value of data is the image that was pressed (and not the whole album) in order for the transition animation to execute properly, otherwise it would animate towards outside the screen (even if we have `initialScrollIndex` set).

Reply:
> Moving (reset! (:data state) messages) inside a use-effect will cause it not to work properly. That way it works properly. Idk if there is a better way to do it.

Unfortunately I can't suggest anything without actually writing code for this :shrug: 

> I believe atoms are initialized outside the renderer.

A *renderer* in Reagent is the function that returns hiccup, it's the result of this function that Reagent interprets. The proper form-2 component requires the renderer to not initialize state, so we need to move the initialization to the outer *layer*.

I think the following is roughly how the code should be (I haven't tested it). There's surely more than one way.

1. Since the state initialization relies on subscription values, move the subscription to the parent component.
2. Create a form-2 component by initializing both `state` and `props` outside the renderer (both are *states* from Reagent's perspective).
3. Avoid the rookie mistake mentioned in Reagent's docs and remember to repeat any needed arguments in the renderer function.

```clojure
(defn- f-lightbox
  [{:keys [messages index]}]
  (let [props (utils/init-props)
        state (utils/init-state messages index)]
    (fn [{:keys [messages index]}]
      (let [animations               (utils/init-animations)
            derived                  (utils/init-derived-animations animations)
            callback                 (fn [e]
                                       (on-viewable-items-changed e props state))]
        (anim/animate (:background-color animations) "rgba(0,0,0,1)")
        (reset! (:data state) messages)
        (utils/orientation-change props state animations)
        (utils/effect props animations index)
        [:f> lightbox-content props state animations derived messages index callback]))))

(defn lightbox
  []
  (let [screen-params (rf/sub [:get-screen-params])]
    [:f> f-lightbox screen-params]))
```

> I added back the comment explaining why

That's nice, thanks @OmarBasem

Reply:
@ilmotta sorry but it's not clear to me what's the benefit of that change. Currently, only `lightbox-content` executes when re-rendering. `lightbox` and `f-lightbox` render only one when the screen is opened.

Reply:
It's not about any benefit or advantage @OmarBasem. I'm just sharing how Reagent should be used according to its documentation.

I get it that it works, but this to me is not good enough. We know many UI bugs over the history of status-mobile happened because Reagent (plus re-frame sometimes) were used incorrectly (in the strict sense of the official docs).

The implementation below works (I'm not showing the diff in the function `lightbox-content`). It's just to show that there are ways to make things relatively simpler *and* follow Reagent's guidelines. I could go on with the refactor, but it's just to give you an idea.

```clojure
(defn- f-lightbox
  [{:keys [messages index]}]
  (let [props (utils/init-props)
        state (utils/init-state messages index)]
    (fn [{:keys [messages index]}]
      (reset! (:data state) messages)
      [:f> lightbox-content props state messages index])))

(defn lightbox
  []
  (let [screen-params (rf/sub [:get-screen-params])]
    [:f> f-lightbox screen-params]))
```

----

I think we spent enough time in this thread @OmarBasem, Up to you to judge the course of action. I think it's reasonable to keep the original code, but in future PRs we can try to be more strictly follow Reagent's guidelines.

Reply:
Thanks for the clarification. Fixed it.